pkg/rules/terraswap/col4_logfinders.go

```diff
@@ -0,0 +1,79 @@
+package terraswap
+
+import (
+	"github.com/dezswap/cosmwasm-etl/pkg/eventlog"
+	"github.com/pkg/errors"
+)
+
+func CreateCreatePairRuleFinder(factoryAddr string) (eventlog.LogFinder, error) {
+	if factoryAddr == "" {
+		errMsg := "no factory address"
+		return nil, errors.New(errMsg)
+	}
+
+	rule := col4CreatePairRule
+	rule.Items[FactoryAddrIdx].Filter = factoryAddr
+
+	return eventlog.NewLogFinder(rule)
+}
+
+func CreatePairCommonRulesFinder(pairs map[string]bool) (eventlog.LogFinder, error) {
+	var filter func(v string) bool
+	if pairs != nil {
+		filter = func(v string) bool {
+			_, ok := pairs[v]
+			return ok
+		}
+	}
+	rule := col4PairCommonRule
+	rule.Items[PairAddrIdx].Filter = filter
+	return eventlog.NewLogFinder(rule)
+}
+
+// Track cw20 transfer
+func CreateWasmCommonTransferRuleFinder(pairs map[string]bool) (eventlog.LogFinder, error) {
```

Comment:
`CreateWasmCommonTransferRuleFinder` and `CreateTransferRuleFinder` seems to have no test yet

Reply:
https://github.com/dezswap/cosmwasm-etl/pull/11/commits/f44accf3054f18fc530d41048dc70b3fbc01a00b